### PR TITLE
Clarify direction in which OrientationMap maps

### LINKED ISSUE #856: Document OrientationMap direction better
The documentation for things using OrientationMaps is vague and sometimes seemingly self-contradictory about which direction the OrientationMap maps.  Neighbors and BlockNeighbor both just call it "the OrientationMap of the neighbor(s)" which does not suggest either direction to me.  The OrientationMap tutorial states that it is the orientation of the block with respect to the neighbor, but then goes on to say that it maps to the neighbor from the block, which seems contradictory.  I assume the last of these (block to neighbor) is correct, because it is the most explicit statement, but the documentation should make this clear.

### PR DESCRIPTION
**Pull request by mar-celine**
## Proposed changes

Fix #856 

### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: docs/Tutorials/Orientation.md [hunk 1]
```diff
@@ -13,14 +13,14 @@ as the logical directions \f$\xi\f$, \f$\eta\f$, and \f$\zeta\f$, where
 domain with multiple elements, the logical directions are not necessarily
 aligned on the interfaces between two elements, as shown in the figure below.
 As fluxes need to be communicated across the boundaries of adjacent elements,
-there needs to be a system in place that organizes the relative orientations of
-elements with respect to their neighbors. This is handled by OrientationMap.
+there needs to be a system in place that catalogs the relative orientations of
```
Comment (kidder) [inline docs/Tutorials/Orientation.md]: As certain operations, (e.g. fluxes, limiting, ...) communicate information across the boundaries of adjacent elements, there needs to be a class that takes into account the relative orientations of neighboring elements.   This class is OrientationMap.

File: src/Domain/BlockNeighbor.hpp [hunk 2]
```diff
@@ -18,18 +18,21 @@ class er;
...
   BlockNeighbor() = default;
 
-  /// Construct with the Id and orientation of the neighbor.
+  /// Construct with the Id and orientation of the neighbor relative to the
+  /// host.
   ///
   /// \param id the Id of the neighbor.
-  /// \param orientation the OrientationMap of the neighbor.
+  /// \param orientation the OrientationMap of the neighbors relative to the
+  /// host. In other words, the OrientationMaps accept objects in the host
```
Comment (kidder) [inline src/Domain/BlockNeighbor.hpp]: This OrientationMap takes objects in the host Block and returns what they are in the neighbor Block.

File: src/Domain/Neighbors.hpp [hunk 3]
```diff
@@ -21,16 +21,19 @@ class ElementId;
...
 class Neighbors {
  public:
-  /// Construct with the ids and orientation of the neighbors.
+  /// Construct with the ids and orientation of the neighbors relative to the
+  /// host.
   ///
   /// \param ids the ids of the neighbors.
-  /// \param orientation the OrientationMap of the neighbors.
+  /// \param orientation the OrientationMap of the neighbors relative to the
+  /// host. In other words, the OrientationMaps accept objects in the host
```
Comment (kidder) [inline src/Domain/Neighbors.hpp]: same as above with Block->Element

**Review by kidder — CHANGES_REQUESTED** [review]

File: src/Domain/BlockNeighbor.hpp (hunk 2 above)
Comment (fmahebert) [reply src/Domain/BlockNeighbor.hpp]: @kidder how about adding "... what they are in the coordinates of the neighbor Block"? I find this phrasing makes it clearer that the same object is viewed in new coordinates.

Comment (kidder) [reply src/Domain/BlockNeighbor.hpp]: @fmahebert  "... takes objects in the logical coordinate frame of the host Block and maps them to the logical coordinate frame of the neighbor Block" ?

Comment (fmahebert) [reply src/Domain/BlockNeighbor.hpp]: Sounds good to me

**Review by wthrowe — COMMENTED** [review]
I don't have anything I want to add to the other reviews.

**Comment by kidder** [comment]
@gsb76 please squash

**Comment by mar-celine** [comment]
@kidder @wthrowe @fmahebert I've squashed :+1: Thanks!

**Review by fmahebert — APPROVED** [review]

**Review by wthrowe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
ignoring travis timeout which occurred after all the tests passed

**Pull request merged**